modules/lang-painless/src/main/java/org/elasticsearch/painless/action/PainlessExecuteAction.java

```diff
@@ -238,29 +237,21 @@ static Tuple<String, String> parseClusterAliasAndIndex(String indexExpression) {
                     return new Tuple<>(null, null);
                 }
                 String trimmed = indexExpression.trim();
-                String sep = String.valueOf(RemoteClusterAware.REMOTE_CLUSTER_INDEX_SEPARATOR);
-                if (trimmed.startsWith(sep) || trimmed.endsWith(sep)) {
-                    throw new IllegalArgumentException(
-                        "Unable to parse one single valid index name from the provided index: [" + indexExpression + "]"
-                    );
-                }
-
+                String[] parts = RemoteClusterAware.splitIndexName(trimmed);
                 // The parser here needs to ensure that the indexExpression is not of the form "remote1:blogs,remote2:blogs"
                 // because (1) only a single index is allowed for Painless Execute and
                 // (2) if this method returns Tuple("remote1", "blogs,remote2:blogs") that will not fail with "index not found".
                 // Instead, it will fail with the inaccurate and confusing error message:
                 // "Cross-cluster calls are not supported in this context but remote indices were requested: [blogs,remote1:blogs]"
                 // which comes later out of the IndexNameExpressionResolver pathway this code uses.
-                String[] parts = indexExpression.split(sep, 2);
-                if (parts.length == 1) {
-                    return new Tuple<>(null, parts[0]);
-                } else if (parts.length == 2 && parts[1].contains(sep) == false) {
-                    return new Tuple<>(parts[0], parts[1]);
-                } else {
+                if ((parts[0] != null && parts[1].isEmpty())
+                    || parts[1].contains(String.valueOf(RemoteClusterAware.REMOTE_CLUSTER_INDEX_SEPARATOR))) {
```

Comment:
I am not sure this is correct, because here originally Painless is not checking whether the second part is the remote index, it just checks if it contains `:` - even if it's not a valid remote index syntax. Basically, any string with more than one `:` is triggering this clause. So passing it to `isRemoteIndexName` would change the semantics here, which I don't want to do unless somebody from the relevant team explicitly approves it. It _might_ be that the original intent was different but in this refactoring I want to keep the original semantics as much as possible to avoid disruptions.

Reply:
I suppose that is fair. The main thing I see different in the logic is that we account for date math when checking if an index is a remote index expression. Is date math supported on CCS? I'm gathering that it is not from the comments, but I might be mistaken or reading too much into the restrictions.

Reply:
Yes, datemath should work for CCS - it'd be either `<datemath>` or `clustername:<datemath>`. We have some tests here: https://github.com/elastic/elasticsearch/blob/main/server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CrossClusterSearchIT.java#L667

Reply:
I will follow up with someone more familiar with Painless and see if date math expressions should be supported here or not. In the mean time, I agree with your assessment to not change the logic.